contracts/CourtRoom.sol

```diff
@@ -7,24 +7,32 @@ import "./ensabstract.sol";
 
 contract CaseContract is Owned {
 
+  struct promise{
+		address beneficiary;
+		uint256 blockNumber;
+		uint8 sig_v;
+		bytes32 sig_r;
+		bytes32 sig_s;
+	}
 	struct claim {
 		bytes32 claimId;
 		address plaintiff;
 		bytes32[] evidence;
+		promise   servicePromise;
 		uint status;
 		bool valid;
```

Comment:
Changing status to an enum could be a neater approach; something like this maybe?
```
enum status {
        Invalid,
        MoreEvidenceRequired,
        UnderReview
}
```
There was the original intention to have a log of all claims that had been opened, valid and invalid ones, to keep track of how many times an address had opened a claim, it was simply a mapping from address => Claim[], thusly creating a basic reputation scheme. But was agreed to be over kill for initial scoping.